engine/src/main/java/org/camunda/bpm/engine/impl/bpmn/parser/BpmnParse.java

```diff
@@ -2646,12 +2646,25 @@ public TaskDefinition parseTaskDefinition(Element taskElement, String taskDefini
     taskDefinition.setKey(taskDefinitionKey);
     processDefinition.getTaskDefinitions().put(taskDefinitionKey, taskDefinition);
 
-    String formKeyAttribute = taskElement.attributeNS(CAMUNDA_BPMN_EXTENSIONS_NS, "formKey");
 
+    String formKeyAttribute = taskElement.attributeNS(CAMUNDA_BPMN_EXTENSIONS_NS, "formKey");
     if (formKeyAttribute != null) {
       taskDefinition.setFormKey(expressionManager.createExpression(formKeyAttribute));
     }
 
+    String formRefAttribute = taskElement.attributeNS(BpmnParse.CAMUNDA_BPMN_EXTENSIONS_NS, "formRef");
+    if(formRefAttribute != null) {
```

Comment:
Should we handle the case when both `formKeyAttribute` and `formRefAttribute` are set? I noticed that we use the `addError(...)` method to handle similar cases in the `BpmnParse`. However, I also saw that we handle this case in the `DefaultFormHandler` class (line 97). What would be the best practice approach?

Reply:
You are right, we handle this use situation in the `DefaultFormHandler`. The reason I decided to do it this way is that for both, TaskForms and StartForms, we parse the form handler configuration. For StartForms the `formKey` is handled within the `DefaultStartFormHandler` which is not the case for TaskForms. So it makes sense to have the necessary checks in the handler as both form types will execute them before setting any values.

I hope this makes sense. Let me know what you think

Reply:
Good point. I like it!